recipes/date/all/conanfile.py

```diff
@@ -1,29 +1,91 @@
-    no_copy_source = True
+    exports_sources = ["patches/*", "CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_system_tz_db": [True, False],
+               "use_tz_db_in_dot": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "use_system_tz_db": True,
+                       "use_tz_db_in_dot": False}
+
+    _cmake = None
 
     @property
     def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+        return "source_subfolder"
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        #requires at least c++11 to work.
+        tools.check_min_cppstd(self, "11")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["ENABLE_DATE_TESTING"] = False
+        cmake.definitions["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
+        cmake.definitions["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
+        cmake.configure()
+
+        self._cmake = cmake
+        return self._cmake
+
+    def requirements(self):
+        if not self.options.use_system_tz_db:
+            self.requires("libcurl/7.67.0")
+
+    def build(self):
         for patch in self.conan_data["patches"][self.version]:
             tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="date.h", dst=os.path.join("include", "date"),
-                  src=os.path.join(self._source_subfolder, "include", "date"))
+        self.copy(pattern="LICENSE.txt", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "CMake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append("pthread")
+
+        if self.options.use_system_tz_db and not self.settings.os == "Windows":
+            use_os_tzdb = 1
+        else:
+            use_os_tzdb = 0
+
+        defines = ["USE_OS_TZDB={}".format(use_os_tzdb)]
+        if self.options.shared:
```

Comment:
```suggestion
        if self.settings.os == "Windows" and self.options.shared:
```

Reply:
Tricky one - but i guess it makes to add the == windows check.
the CMakeLists.txt on v2.4.1 does not provide the USE_DLL define at all.
but one on master does, and it has a check for win32. 
will add.
https://github.com/HowardHinnant/date/blob/master/CMakeLists.txt#L111

Reply:
applied.

Reply:
sorry, now its ammended in. missed this somehow.